lib/web3/contract.js

```diff
@@ -129,10 +116,10 @@ var checkForContractAddress = function(contract, abi, callback){
 
             } else {
 
-                web3.eth.getTransactionReceipt(contract.transactionHash, function(e, receipt){
+                contract._web3.eth.getTransactionReceipt(contract.transactionHash, function(e, receipt){
                     if(receipt && !callbackFired) {
 
-                        web3.eth.getCode(receipt.contractAddress, function(e, code){
+                        contract._web3.eth.getCode(receipt.contractAddress, function(e, code){
```

Comment:
you call it here `_web3` but in the `ContractFactory` `this.web3`, are you sure thats correct? (I might miss something..)